core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -85,9 +93,14 @@ public void run(final ActionContext context) throws Exception {
 
       LOG.debug("Output: {}", out);
       LOG.error("Errors: {}", err);
+      Integer exitCode = session.getExitStatus();
+      if (exitCode != null && exitCode > 0) {
+        throw new IOException(String.format("Error: command finished with exit code: %d", exitCode));
+      }
 
       if (config.output != null) {
-        context.getArguments().set(config.output, out);
+        //regex removes the carriage return at the end of the line
+        context.getArguments().set(config.output, out.replaceAll("\\s+$", ""));
```

Comment:
the ssh library always adds a newline at the end of stdout? This will actually do more than just remove a newline, it will remove all whitespace from the end stdout, including any actual whitespace returned by the command. 

If we know the library always appends a newline, you can just check that out ends with a newline and do a substring.